test/image_test.py

```diff
@@ -561,37 +560,35 @@ def RotateARGBtoRGBA(str_buf):
             test_surface, test_surface, "failing with identical surfaces"
         )
 
-        rgba_buf = pygame.image.tostring(test_surface, "RGBA")
-        rgba_buf = RotateARGBtoRGBA(RotateRGBAtoARGB(rgba_buf))
-        test_rotate_functions = pygame.image.fromstring(
-            rgba_buf, test_surface.get_size(), "RGBA"
-        )
-
-        self._assertSurfaceEqual(
-            test_surface, test_rotate_functions, "rotate functions are not symmetric"
-        )
-
-        rgba_buf = pygame.image.tostring(test_surface, "RGBA")
-        argb_buf = RotateRGBAtoARGB(rgba_buf)
-        test_from_argb_string = pygame.image.fromstring(
-            argb_buf, test_surface.get_size(), "ARGB"
-        )
+        for pair in fmt_combinations:
+            rgba_buf = pygame.image.tostring(test_surface, pair[0])
+            rgba_buf = convert(pair[1], pair[0], convert(pair[0], pair[1], rgba_buf))
+            test_rotate_functions = pygame.image.fromstring(
+                rgba_buf, test_surface.get_size(), pair[0]
+            )
 
-        self._assertSurfaceEqual(
-            test_surface, test_from_argb_string, '"RGBA" rotated to "ARGB" failed'
-        )
+            self._assertSurfaceEqual(
+                test_surface,
+                test_rotate_functions,
+                "converting {} to {} and back is not symmetric".format(
+                    pair[0], pair[1]
+                ),
+            )
 
-        argb_buf = pygame.image.tostring(test_surface, "ARGB")
-        rgba_buf = RotateARGBtoRGBA(argb_buf)
-        test_to_argb_string = pygame.image.fromstring(
-            rgba_buf, test_surface.get_size(), "RGBA"
-        )
+        for pair in fmt_permutations:
```

Comment:
I'm confused about the difference between this test loop and the test loop above, which is for format combinations. I see they use different lists, and the top one has an additional convert step.

Inside the loops they have variable names like `rgba_buf`, which seems misleading, because I'm pretty sure they won't always be RGBA.

Reply:
Apologies.
Yes, the naming is now confusing. 
Those were the previous names that made sense when there was only 2 formats.
I was focusing on the code working and not breaking anything.
Forgot to change the names to reflect the multiple formats.

In terms of perms vs combs:
In the combinations loop there is a 2 way conversion. Permutations here would be redundant.
The permutations loop is only performing a 1 way conversion.

Reply:
The new variables names help the understandability a lot, nice!